include/residuals.h

```diff
@@ -8,6 +8,13 @@
 
 namespace GLnexus {
 
+// Dataset and the records it has for a particular site
+struct DatasetSiteInfo {
+    std::string name;
+    std::shared_ptr<const bcf_hdr_t> header;
+    std::vector<std::shared_ptr<bcf1_t>> records;
+};
+
 class Residuals {
```

Comment:
Yeah I think it would be an overall simplification to change the Residuals class into a `gen_residuals` function and then just pass around `ResidualsFile` instead of both `ResidualsFile` and `Residuals`. Please do, then this LGTM to squash merge. I really like how this PR is simplifying the code overall (and should even moreso when we unify the residuals and loss tracking)